### PR TITLE
Added support for multipart

### PR DESCRIPTION
**Pull request by jmango22**
Added support for multipart proxying.

The best I could do was Local Request File -> Local Disk File -> Remote Request File -> Remote Response -> Local Response.

If there is a better way to do this, I would appreciate some feedback/examples.

Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@domoinc/ryuu-proxy",
-  "version": "3.0.2",
+  "version": "3.0.3",
```
Comment (jpumford) [inline package.json]: Any reason we aren't making this a feature release?

File: src/lib/Transport/index.ts [hunk 2]
```diff
@@ -49,6 +48,17 @@ export default class Transport {
     );
   }
 
+  isMultiPartRequest(headers: IncomingHttpHeaders): boolean {
+    for (const header of Object.keys(headers)) {
```
Comment (jpumford) [inline src/lib/Transport/index.ts]: How about this?
```
   Object.entries(headers).some(([header, value]) => header.toLowerCase() === 'content-type' && value.toString().toLowerCase().includes('multipart'))
```

File: src/lib/Transport/index.ts [hunk 3]
```diff
@@ -101,6 +111,17 @@ export default class Transport {
   }
 
   build(req: IncomingMessage): Promise<request.Options> {
+    return this.buildBasic(req)
+      .then((options) => {
+        return this.parseBody(req).then((body) => {
```
Comment (jpumford) [inline src/lib/Transport/index.ts]: You could flatten this promise chain

File: src/lib/Transport/index.ts [hunk 4]
```diff
@@ -135,13 +149,27 @@ export default class Transport {
       ? (`${headers.referer}&context=${context}`)
       : (`${headers.referer}?userId=27&customer=dev&locale=en-US&platform=desktop&context=${context}`);
 
-    const newHeaders = {
+    if (this.isMultiPartRequest(headers)) {
+      // remove properties that are provided by the form-data library in request
+      return {
+        ...Object.keys(headers).reduce(
```
Comment (jpumford) [inline src/lib/Transport/index.ts]: Nice!

File: package.json (hunk 1 above)
Comment (jmango22) [reply package.json]: I see it kind of like a bug fix. The proxy should be able to handle requests. It was failing on multipart requests, and I fixed it. If you see it as more as a new feature I could change it!

File: src/lib/Transport/index.ts (hunk 4 above)
Comment (jmango22) [reply src/lib/Transport/index.ts]: haha, I was debating just using the `del` operator, but I knew that you would be upset if I made the headers mutable.

File: src/lib/Transport/index.ts (hunk 2 above)
Comment (jmango22) [reply src/lib/Transport/index.ts]: Cool idea. I'll put it in!

**Comment by jmango22** [comment]
@jpumford Can get another review on this?

File: package.json (hunk 1 above)
Comment (jpumford) [reply package.json]: No, I think that's a reasonable explanation!

File: src/lib/Transport/index.ts (hunk 2 above)
Comment (jpumford) [reply src/lib/Transport/index.ts]: Haha maybe it's just harder to read now.

Oh well!

**Pull request merged**